src/Symfony/Component/HttpKernel/EventListener/ResponseListener.php

```diff
@@ -46,6 +48,11 @@ public function onKernelResponse(ResponseEvent $event)
             $response->setCharset($this->charset);
         }
 
+        if ($this->enabledLocales && !$response->isInformational() && !$response->isEmpty() && !$response->headers->has('Content-Language')) {
+            $response->headers->set('Content-Language', $event->getRequest()->getLocale());
+            $response->setVary('Accept-Language', false);
```

Comment:
This needs special handling in case `Vary` is already set (which is usually the case with API Platform). To be honest, I'm not sure that it will be easy to automate, maybe should we let this task to the user (API Platform already provides a configuration option for this header).

Also, this header should be set only if there is more than one enabled language.

In my opinion, this option should be opt-in as it may allow bypassing the cache (and so facilitate DOS attacks), and maybe useless in some cases (e.g. when passing the locale as part of the URL instead of in an HTTP header).

Reply:
oh, I missed this Vary rule when requesting the feature being opt-in. But this indeed makes it a lot more important to make it opt-in (and not even opt-out)

Reply:
> This needs special handling in case Vary is already set 

As discussed privately, the 2nd parameter of `Response::setVary()` controls whether the vary headers should be replaced. When it's passed false like here, a merge is applied so unless I'm mistaken there is no need for additional checks, an existing vary won't get replaced.

> Also, this header should be set only if there is more than one enabled language.

Makes sense, done.

> But this indeed makes it a lot more important to make it opt-in

It's now fully opt-in via a new `use_language_headers` option.